internal/pushgateway/client.go

```diff
@@ -0,0 +1,66 @@
+package pushgateway
+
+import (
+	"bytes"
+	"context"
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/opentracing/opentracing-go"
+)
+
+type httpClient struct {
+	hc *http.Client
+}
+
+// NewHTTPClient creates a new Push Gateway client.
+func NewHTTPClient(disableTLSValidation bool) Client {
+	hc := &http.Client{
+		Timeout: 30 * time.Second,
+		Transport: &http.Transport{
+			DisableKeepAlives: true,
```

Comment:
IIRC, the SyTest HTTP server was flaky when a connection was reused. It's especially fragile if a test fails without cleaning up. The HTTP server then becomes confused about which parts belong to the next test.

Originally, this was set to `disableTLSValidation`, because it's used only for testing. Not sure why it slipped out of that.